### PR TITLE
Validate identifiers in spreadsheet before uploading file

### PR DESCRIPTION
**Pull request by smrohrer**
If a spreadsheet for bulk upload includes an invalid identifier, the CLI would upload the entire file before reporting the error, and the entire file would have to be uploaded again after fixing the identifier.

This was previously noted in #203.

The CLI will now validate the identifier on the client side before uploading, and will terminate the program if an invalid identifier is encountered.

### CODE REVIEW

File: internetarchive/cli/ia_upload.py [hunk 1]
```diff
@@ -265,6 +265,15 @@ def main(argv, session):
                 if not identifier:
                     print('error: no identifier column on spreadsheet!')
                     sys.exit(1)
+                try:
+                    validate_ia_identifier(identifier)
+                except:
```
Comment (cclauss) [inline internetarchive/cli/ia_upload.py:270]: PEP8 and https://realpython.com/the-most-diabolical-python-antipattern document why bare exceptions should be avoided.

Comment (smrohrer) [reply internetarchive/cli/ia_upload.py]: Thanks for pointing that out

**Comment by jjjake** [comment]
Thanks for the contribution!

I think this should be an option (off by default) in case the rules for a valid identifier shift, and I think the option should work for both spreadsheet mode and regular uploads. I also think that the `_validate_identifier` function should be moved to `internetarchive.utils.validate_identifier`, and implemented in `internetarchive.upload_file` and `internetarchive.upload` (and turned on in `ia_upload.py` in the Item.upload method calls). Perhaps it should be named something like `validate_s3_identifier()` as well, because there are different rules for what is considered a valid identifier at item creation vs. modifying metadata, for example.

Please let me know if you have any questions. I think this would be a helpful feature to help deal with the `requests` module lack of support for 100-continue (this is what requires the whole file to be uploaded before an error is returned by IA-S3).

**Comment by smrohrer** [comment]
Thank you Jake for the feedback! I will push up updates shortly.

To be clear, do you want the CLI to not validate identifiers by default, both for spreadsheet and regular uploads? Currently for regular uploads via the CLI, the identifier is always validated as part of the arg scheme validation.

Thanks!

**Comment by jjjake** [comment]
Hey @smrohrer, sorry for the confusion and thank you for the contribution. I deleted my comment after posting because I realized I needed to take a closer look at the current code. I forgot that we're already validating for non-spreadsheet-mode uploads by default.

With that in mind, your changes look great. I would just suggest that we _do_ validate by default to be consistent with the existing behavior. I don't think we need a command-line option to toggle the validation in this case, since it's been the default for years without issue.

TL;DR, your changes look great. Let's just remove the `--validate-identifier` option, and validate identifier's by default. Let me know if you have any questions or thoughts on this. Thanks again!

**Comment by smrohrer** [comment]
Thank you Jake for reviewing, glad I can help make the tool a little better!

**Comment by jjjake** [comment]
Looks great, @smrohrer, thank _you_! Thanks for writing tests as well! :)
